### PR TITLE
Connect apis

### LINKED ISSUE #2: Give category name unique keys
Each category in the dropdown needs to have it's own key in React

### LINKED ISSUE #3: Get categories from API in dropdown menu
As a developer I want to be able to display all the categories from the API in a dropdown menu. 
This comes from the API so as new categories are added or removed the dropdown should be accurate.

### LINKED ISSUE #4: Get random API rendering in search results section
As a User when I press the "Get Random API" button I expect to see a random api listing displayed on the page.

### LINKED ISSUE #5: Build accumulative search method
As a developer I want to be able to take all the values from the search dropdowns and return all relevant results to my user. I'm calling this an accumulative search.

This would work also with the entries API call.

Currently there is are options of "any" for each field which would return all apis(649) - may need to prevent/rework that.

### LINKED ISSUE #6: Add Router to application
As a user I would like to be able to navigate between pages with forward and back buttons.
As a developer I would like to conditionally render components based on the URL.

### LINKED ISSUE #8: Render search results conditionally
As a developer I want to be able to render the search results as card if they come from the drop-downs or the random button. These two display should not compete.

### LINKED ISSUE #9: Get DetailCards to display
As a user when I search by category or by random, I would like to be able to see results displayed on the page.

### PR DESCRIPTION
**Pull request by Estaffieri**
### What does this PR do?
This PR builds out the following components:
- App
- Home
- Favorites
- SearchResults
- SearchForm
- DetailCard

Here I connected and used data from the apiCalls to populate the search form, generate a random api, and display the correct cards for each search.

Router is also present in this PR within the App component. Only switching between the Home and Favorites views.

In earlier commits the search form was far more complex, but due to frequent missing data values within the PublicAPis API I decided that the best way forward was to simplify. My search is now only categories and allows the user to browse all options that might be of interest. This removed some heavy conditional logic and error handling for variable or missing values in a public api.

While the favorites component is present and some empty methods are living in App this functionality isn't yet included in this project.

### How should this be tested?
Clone down repo
run '''npm start'''
Select a category from Category drop-down and click "Search" button.
Select the "Get Random Api" button.

### CODE REVIEW

File: src/Components/Home/Home.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            categories: [],
+            focus: {}
+        }
+
+    }
+
+    componentDidMount() {
+
+        this.setState(getAllCategories())
+        console.log(this.state.categories)
```
Comment (jmevans0211) [inline src/Components/Home/Home.js]: @Estaffieri is this console logging as intended?

Comment (Estaffieri) [reply src/Components/Home/Home.js]: This was experimental and should have been removed prior to pushing.

**Comment by Estaffieri** [comment]
In its current condition this app won't load due to an error in Search.addCategoriesForSearch() where we have a TypeError: Cannot read property "map" of undefined.

I converted all the apicalls to async and expanded the search form.

File: src/Components/Home/Home.js (hunk 1 above)
Comment (Estaffieri) [reply src/Components/Home/Home.js]: I have removed this console.log.

**Pull request merged**